From 37ead19eb0a523cf0eff8971c87e4f67a5dc834b Mon Sep 17 00:00:00 2001
From: marcmyworld <moviesminister@gmail.com>
Date: Sat, 12 Sep 2026 05:03:12 +0530
Subject: [PATCH 1/2] Fix Movie Mode preview freeze by deferring
 setInputSurface to prepare()

In Android 16, calling MediaRecorder.setInputSurface() before setVideoSource()
and setOutputFormat() throws an IllegalStateException. Defer setInputSurface()
until prepare() when the recorder is properly initialized.
---
 smali_classes4/xb/w.smali | 29 ++++++++++++++++++++++++++---
 1 file changed, 26 insertions(+), 3 deletions(-)

diff --git a/smali_classes4/xb/w.smali b/smali_classes4/xb/w.smali
index 09f7382c..95f3a3c7 100644
--- a/smali_classes4/xb/w.smali
+++ b/smali_classes4/xb/w.smali
@@ -9,6 +9,8 @@
 # instance fields
 .field public final a:Landroid/media/MediaRecorder;
 
+.field public b:Landroid/view/Surface;
+
 
 # direct methods
 .method public constructor <init>()V
@@ -408,9 +410,7 @@
 .method public final e(Landroid/view/Surface;)V
     .locals 0
 
-    iget-object p0, p0, Lxb/w;->a:Landroid/media/MediaRecorder;
-
-    invoke-virtual {p0, p1}, Landroid/media/MediaRecorder;->setInputSurface(Landroid/view/Surface;)V
+    iput-object p1, p0, Lxb/w;->b:Landroid/view/Surface;
 
     return-void
 .end method
@@ -515,6 +515,29 @@
         }
     .end annotation
 
+    iget-object v0, p0, Lxb/w;->b:Landroid/view/Surface;
+
+    if-eqz v0, :cond_0
+
+    :try_start_0
+    iget-object v1, p0, Lxb/w;->a:Landroid/media/MediaRecorder;
+
+    invoke-virtual {v1, v0}, Landroid/media/MediaRecorder;->setInputSurface(Landroid/view/Surface;)V
+    :try_end_0
+    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
+
+    goto :cond_0
+
+    :catch_0
+    move-exception v0
+
+    const-string v1, "CED_MiMediaRecorder"
+
+    const-string v2, "failed to setInputSurface"
+
+    invoke-static {v1, v2, v0}, Lcom/android/camera/log/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)V
+
+    :cond_0
     invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
 
     move-result-wide v0
-- 
2.55.0

